sdk/eventhub/Azure.Messaging.EventHubs/src/PartitionEvent.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Core;
+
+namespace Azure.Messaging.EventHubs
+{
+    /// <summary>
+    ///   Contains information about a partition that has attempted to receive an event from the Azure Event Hub
+    ///   service, as well as the received event, if any.
+    /// </summary>
+    ///
+    public class PartitionEvent
```

Comment:
Could we please make this a struct, if you would be so kind.   That's going to require a public constructor, which is a reasonable trade-off.